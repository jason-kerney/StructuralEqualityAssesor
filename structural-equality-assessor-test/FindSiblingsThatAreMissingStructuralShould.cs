using System;
using System.Collections.Generic;
using System.Linq;
using ApprovalTests;
using StructuralEqualityAssessor.Test.Examples;
using StructuralEqualityAssessor.Test.Examples.LookUpFailingClasses;
using StructuralEqualityAssessor.Test.Examples.LookUpGoodClasses;
using StructuralEqualityAssessor.Test.Examples.LookUpIgnoredClasses;
using StructuralEqualityAssessor.Test.Examples.LookUpMixedClasses;
using Xunit;

namespace StructuralEqualityAssessor.Test
{
    public class FindSiblingsThatAreMissingStructuralShould
    {
        [Fact]
        public void NotFindAnythingIfAllAreCorrect()
        {
            var results = FindSiblingsThatAreMissingStructural.Equality<HasSimpleArrays>();
            Assert.Empty(results);
        }

        [Fact]
        public void FindAllWrongWhenAllAreWrong()
        {
            var results = FindSiblingsThatAreMissingStructural.Equality<AnotherClass>();

            var friendlyResults = results.Select(t => t.ToString()).OrderBy(t => t);
            Approvals.VerifyAll(friendlyResults, "Bad Equality");
        }

        [Fact]
        public void IgnoreAllIgnoredTypes()
        {
            var result = FindSiblingsThatAreMissingStructural.Equality<ReferenceGoodClass>();
            Assert.Empty(result);
        }

        [Fact]
        public void OnlyReportFailingTypesIfMixed()
        {
            var ctors = new Dictionary<Type, Func<object>>
            {
                {typeof(NonDefaultConstructor), () => new NonDefaultConstructor(44)} 
            };
            var results = FindSiblingsThatAreMissingStructural.Equality<BadEqualityClass>(null, ctors);
            var friendlyResults = results.Select(t => t.ToString()).OrderBy(t => t);
            Approvals.VerifyAll(friendlyResults, "Bad Equality");
        }
    }
}