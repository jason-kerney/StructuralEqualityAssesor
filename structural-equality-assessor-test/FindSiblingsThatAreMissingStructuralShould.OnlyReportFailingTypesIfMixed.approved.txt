Bad Equality[0] = StructuralEqualityAssessor.Test.Examples.LookUpMixedClasses.BadEqualityClass
